packages/ozone/src/tag-service/embed-content-type-tagger.ts

```diff
@@ -0,0 +1,45 @@
+import { AppBskyEmbedImages } from '@atproto/api'
+import { langLogger as log } from '../logger'
+import { ContentTagger } from './content-tagger'
+
+export class EmbedContentTypeTagger extends ContentTagger {
+  tagPrefix = 'embed-content-type:'
+  isApplicable(): boolean {
+    return Boolean(
+      this.subjectStatus &&
+        this.subject.isRecord() &&
+        this.subject.uri.includes('app.bsky.feed.post') &&
+        !this.subjectStatus.tags?.find((tag) => tag.includes(this.tagPrefix)),
+    )
+  }
+
+  async getTags(): Promise<string[]> {
+    try {
+      const recordValue = await this.getRecordValue()
+      if (!recordValue) {
+        return []
+      }
+      const tags: string[] = []
+      if (AppBskyEmbedImages.isView(recordValue)) {
+        tags.push(`${this.tagPrefix}image`)
+      }
```

Comment:
If `recordValue` is a record then we don't have our hands on a view (`isView()`).  I'd expect something roughly along these lines:
```suggestion
      if (
        AppBskyFeedPost.isRecord(recordValue) &&
        (AppBskyEmbedImages.isMain(recordValue.embed) ||
          AppBskyEmbedRecordWithMedia.isMain(recordValue.embed))
      ) {
        tags.push(`${this.tagPrefix}image`)
      }
```

Reply:
thanks. was gonna ask for the correct check here once we have the video stuff available on main.